cbauthx/cbauth_test.go

```diff
@@ -108,3 +112,228 @@ func TestCbAuthBasicSlow(t *testing.T) {
+		UserAgent: "stg",
+		Endpoint:  "http://" + testutils.TestOpts.HTTPAddrs[0],
+		Username:  testutils.TestOpts.Username,
+		Password:  testutils.TestOpts.Password,
+	}
+
+	// check that user1 is handled properly before creation
+	_, err = auth.CheckUserPass(ctx, user1, "password")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	_, err = auth.CheckUserPass(ctx, user1, "badpassword")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	// create user1 and user2
+	err = mgmt.UpsertUser(ctx, &cbmgmtx.UpsertUserOptions{
+		Username:    user1,
+		DisplayName: user1,
+		Password:    "password1",
+		Roles:       []string{"ro_admin"},
+	})
+	require.NoError(t, err)
+
+	err = mgmt.UpsertUser(ctx, &cbmgmtx.UpsertUserOptions{
+		Username:    user2,
+		DisplayName: user2,
+		Password:    "password2",
+		Roles:       []string{"ro_admin"},
+	})
+	require.NoError(t, err)
+
+	// check that user1 is handled properly after creation
+	_, err = auth.CheckUserPass(ctx, user1, "password1")
+	assert.NoError(t, err)
+
+	_, err = auth.CheckUserPass(ctx, user1, "badpassword1")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	// check that user2 is handled properly after creation
+	_, err = auth.CheckUserPass(ctx, user2, "password2")
+	assert.NoError(t, err)
+
+	_, err = auth.CheckUserPass(ctx, user2, "badpassword2")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	// delete user1 and user2
+	err = mgmt.DeleteUser(ctx, &cbmgmtx.DeleteUserOptions{
+		Username: user1,
+	})
+	require.NoError(t, err)
+
+	err = mgmt.DeleteUser(ctx, &cbmgmtx.DeleteUserOptions{
+		Username: user2,
+	})
+	require.NoError(t, err)
+
+	// user1 and user2 should both fail with bad passwords before propagation
+	_, err = auth.CheckUserPass(ctx, user1, "badpassword1")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	_, err = auth.CheckUserPass(ctx, user2, "badpassword2")
+	assert.ErrorIs(t, err, ErrInvalidAuth)
+
+	// user1 should propagate and be invalidated pretty quickly
+	assert.Eventually(t, func() bool {
+		_, err = auth.CheckUserPass(ctx, user1, "password1")
+		return err == ErrInvalidAuth
+	}, 3*time.Second, 100*time.Millisecond)
+
+	// user2 should propagate and be invalidated pretty quickly
+	assert.Eventually(t, func() bool {
+		_, err = auth.CheckUserPass(ctx, user2, "password2")
+		return err == ErrInvalidAuth
+	}, 3*time.Second, 100*time.Millisecond)
+
+	// clean up
+	_ = auth.Close()
+}
+
+func TestCbauthFailuresDino(t *testing.T) {
+	testutils.SkipIfNoDinoCluster(t)
+	testutils.SkipIfOlderServerVersion(t, "7.2.0")
+
+	ctx := context.Background()
+	logger := testutils.MakeTestLogger(t)
+	nodes := testutils.NsGetNodes(t)
+
+	orchEndpoint := testutils.NsGetOrchestratorEndpoint(t)
+	blockNode := nodes.PickFirst(t, []string{orchEndpoint})
+	blockHost := blockNode.Hostname
+	blockNsEndpoint := blockNode.NsEndpoint()
+
+	var allEndpoints []string
+	for _, node := range nodes {
+		allEndpoints = append(allEndpoints, node.NsEndpoint())
+	}
+
+	log.Printf("orchestrator: %s", orchEndpoint)
+	log.Printf("nodes:")
```

Comment:
I assume these printfs were left here deliberately?

Reply:
Yea, they help when debugging after the test fails.